src/ucs/vfs/base/vfs_obj.c

```diff
@@ -293,8 +308,13 @@ static ucs_status_t ucs_vfs_node_add(void *parent_obj, ucs_vfs_node_type_t type,
         token = strsep(&next_token, "/");
     }
 
-    ucs_vfs_node_build_path(current_node, token, abs_path_buf,
-                            sizeof(abs_path_buf));
+    if (type == UCS_VFS_NODE_TYPE_DIR) {
+        ucs_vfs_node_build_unique_path(current_node, token, obj, abs_path_buf,
+                                       sizeof(abs_path_buf));
+    } else {
+        ucs_vfs_node_build_path(current_node, token, abs_path_buf,
```

Comment:
seems we can always call ucs_vfs_node_build_path (for both file and dir), then adjust the path for "dir" if needed